packages/veritone-widgets/src/widgets/FolderSelectionDialog/NewFolder.js

```diff
@@ -0,0 +1,234 @@
+    open: bool,
+    createFolder: func,
+    selectFolder: func,
+    selectedFolder: shape({
+      id: string,
+      treeObjectId: string,
+      orderIndex: number,
+      name: string,
+      description: string,
+      modifiedDateTime: string,
+      status: string,
+      ownerId: string,
+      typeId: number,
+      parent: shape({
+        treeObjectId: string
+      }),
+
+      childFolders: shape({
+        count: number
+      })
+    }),
+    rootFolder: shape({
+      id: string,
+      name: string,
+      treeObjectId: string,
+      organizationId: string,
+      ownerId: string,
+      typeId: number,
+      orderIndex: number,
+      childFolders: shape({
+        count: number
+      })
+    }),
+    newFolder: shape({
+      loading: bool,
+      error: bool,
+      errorMessage: string,
+      folder: shape({
+        id: string,
+        treeObjectId: string,
+        orderIndex: number,
+        name: string,
+        description: string,
+        modifiedDateTime: string,
+        status: string,
+        parent: shape({
+          treeObjectId: string
+        }),
+
+        childFolders: shape({
+          count: number
+        })
+      })
+    }),
+    cancel: func,
+    resetNewFolder: func
+  };
+
+  static defaultProps = {
+    open: false
+  };
+
+  state = {
+    newFolderName: ''
+  };
+
+  componentDidMount() {
+    const {
+      selectFolder,
+      selectedFolder,
+      rootFolder,
+      resetNewFolder
+    } = this.props;
+    // when opening newfolder dialog - default the folder selected to the root folder if they did not select one
+    if (!selectedFolder.treeObjectId) {
+      selectFolder(rootFolder);
+    }
+    resetNewFolder();
+  }
+
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    const { cancel } = this.props;
+    // we don't want to close the new folder dialog if there is an error creating the new folder
+    if (
+      isEmpty(prevProps.newFolder.folder) &&
+      !isEmpty(this.props.newFolder.folder)
+    ) {
+      cancel();
+    }
+  }
+
+  handleCancel = () => {
+    // close the dialog
+    const { cancel } = this.props;
+    cancel();
+  };
+
+  onChange = event => {
+    // set state for name of new folder -  prevents typing an empty space to begin
+    let name = event.target.value.replace(/^\s+/g, '');
+    this.setState({
+      newFolderName: name
+    });
+  };
+
+  createNewFolder = () => {
+    // creates new folder  - checks for empty name - if no folder name - we resetNewFolder with error to alert them
+    // otherwise we create new folder
+    // from redux createFolder(name, description, parentId, orderIndex, appType, folder) the other fields like description
+    // and order index are here if we ever want to expand this
+    const { createFolder, selectedFolder, resetNewFolder } = this.props;
+    if (this.state.newFolderName) {
+      createFolder(
+        this.state.newFolderName,
+        '',
+        selectedFolder.id,
+        0,
+        'cms',
```

Comment:
Can we pass this in as a prop?  The widget shouldn't be specific to "cms" rootFolderType.